build/pmm-client-docker/entrypoint.py

```diff
@@ -18,21 +18,36 @@
 from __future__ import print_function, unicode_literals
 import os
 import subprocess
+import signal
 import sys
 import time
 from distutils.util import strtobool
 
-
 PMM_AGENT_SETUP            = strtobool(os.environ.get('PMM_AGENT_SETUP', 'false'))
+PMM_AGENT_SIDECAR          = strtobool(os.environ.get('PMM_AGENT_SIDECAR', 'false'))
+PMM_AGENT_SIDECAR_SLEEP    = int(os.environ.get('PMM_AGENT_SIDECAR_SLEEP', '1'))
 PMM_AGENT_PRERUN_FILE      = os.environ.get('PMM_AGENT_PRERUN_FILE', '')
 PMM_AGENT_PRERUN_SCRIPT    = os.environ.get('PMM_AGENT_PRERUN_SCRIPT', '')
 
 
+class GracefulKiller:
+    process = None
+
+    def __init__(self):
+        signal.signal(signal.SIGINT, self.exit_gracefully)
+        signal.signal(signal.SIGTERM, self.exit_gracefully)
```

Comment:
According to this doc, listening to SIGTERM is pretty much useless as the pod will be terminated right away. https://cloud.google.com/blog/products/containers-kubernetes/kubernetes-best-practices-terminating-with-grace
Probably no harm to do like this, but also good to acknowledge.

Frankly, this PR would benefit from good comments as the implementation may not be readily digestible by the reader (like it was in my case). We will revisit this PR many times, and we would just save ourselves time if we get it properly documented.

Reply:
Where do you see that we are listening to SIGKILL?

Reply:
SIGINT, I mistyped.

Reply:
I'm just reading the docs, I hope they are recent.

Reply:
that's useless in context of kubernetes, but pmm-client is also can be used as standalone docker container which can be stopped using Ctrl+C(SIGINT), we can't just ignore that.

Reply:
,Maybe. I thought the term 'sidecar' is only applicable to k8s. But that's even more reason to talk about it in the code comments, no?

Reply:
Then we should ignore that case?
No, there are only 110 lines of code where we have a lot of `print`s which explains what we are doing there. What does `GracefullKiller` explains it's name.

Reply:
And regarding SIGINT, it's required even without SIDECAR env variable.

Reply:
Yeah, I agree - SIGINT should stay.

Reply:
We already heave three use cases here:
1. pmm-client container running in k8s
2. pmm-client container running in a non-k8s env with SIDECAR mode ON
3. pmm-client container running in a non-k8s env with SIDECAR mode OFF (default)

This code needs proper documentation. And more the users who would benefit from it, not only us.